Core/NoOpRunner.Core/Iterators/IteratorAggregate.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace NoOpRunner.Core.Iterators
+{
+    public abstract class IteratorAggregate : IEnumerable
```

Comment:
I don't think they'll like you extending `IEnumerable` here, as we are supposed to implement the pattern from scratch. Even though you still define all the required methods for the pattern. But eh, maybe we can justify this, saying we needed it for nice froeach loops blah blah..

Reply:
Implemented Iterator pattern according to this:
https://refactoring.guru/design-patterns/iterator/csharp/example

Reply:
I'm not saying it's not a valid impl, in a real scenario you would of course want to use the interfaces. Still, the prof might not like it. But we can just leave it as-is for now, and you can change it if the professor actually complains